### DISCUSSION TITLE
Weakness to maintain asynchronous logic

### DISCUSSION THREAD

**Discussion by baeharam** [mobxjs/mobx-state-tree#1985] [Q&A] [answered]
MST is state management library, but it doesn't solve challenges caused from asynchronous requests. Therefore, I think it would be better to separate async logic with [query](https://tanstack.com/query/v4). We can solve problems on our own, but it's verbose and could be buggy.

You can read [Migrating from MobX (State-Tree) to React Query](https://www.andronio.me/2021/11/15/migrating-from-mobx-state-tree-to-react-query) and will figure out the problems related to async server requests.

So, is it better approach to extract async logic into query library? I couldn't find out other way and out codebase is big enough.

Thanks! 😄

**Comment by EmilTholin** [comment, marked as answer]
Hi @baeharam!

I think it's hard to give a blanket answer for all use cases. Using just MST and having your asynchronous logic in `flow` actions works great for a lot of apps, but just treating MST like a store and having more complex fetch logic in your React components with React Query is a nice way to go about it as well, if your particular project demands it.
